torch/_inductor/codegen/triton.py

```diff
@@ -2236,19 +2240,41 @@ def codegen_node_schedule(self, node_schedule, numel, reduction_numel):
             node_schedule, numel, reduction_numel
         )
 
+        kernel_args = tiled_groups
+        kernel_kwargs = {
+            "reduction_hint": reduction_hint_val,
+            "mutations": mutations,
+            "index_dtype": index_dtype,
+        }
+        V.graph.save_state()
         kernel = TritonKernel(
-            *tiled_groups,
-            reduction_hint=reduction_hint_val,
-            mutations=mutations,
-            index_dtype=index_dtype,
+            *kernel_args,
+            **kernel_kwargs,
         )
 
         self.codegen_node_schedule_with_kernel(node_schedule, kernel)
 
         src_code = kernel.codegen_kernel()
         kernel_name = self.define_kernel(src_code, node_schedule)
+        kernel.kernel_name = kernel_name
+
+        if kernel.persistent_reduction and config.triton.multi_kernel:
+            V.graph.restore_state()
```

Comment:
Can this be done without checkpoint/restore on graph state?

Reply:
That will requires the kernel codegen to be stateless? But we access/update quite a few states during kernel codegen so far:
- allocated/freed buffers
- the generated wrapper code

etc.